src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java

```diff
@@ -353,7 +353,12 @@ private void setDescriptionFields(FilingApi filing, Transaction transaction) {
       filing.setDescriptionIdentifier(filingDescriptionIdentifier.toUpperCase());
     }
     if(filingDescription != null) {
-      filing.setDescription(filingDescription.replace("{date}", formattedDate).toUpperCase());
+      if (transaction.getCompanyNumber() != null){
+        filingDescription = filingDescription + " LIMITED";
+        filing.setDescription(filingDescription.replace("{date}", formattedDate).toUpperCase());
```

Comment:
You dont need else here. If companyNumber is there then add LIMITED in description otherwise it remains as it is. And then setDescription line can live outside if.